viewing_party/party.py

```diff
@@ -1,23 +1,180 @@
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    unique = []
+    total_movies_list = []
+    users_movies = []
+    friends_movies = []
+
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+
+    total_movies_list = users_movies + friends_movies
+
+    for item in total_movies_list:
+        if item not in friends_movies:
+            unique.append(item)
+    return unique
+
+def get_friends_unique_watched(user_data):
+    friends_unique = []
+    users_movies = []
+    friends_movies = []
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in friends_unique:
+                friends_unique.append(item)
+    return friends_unique
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommended = []
+    users_movies = []
+    friends_movies = []
+
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in recommended:
+                if item["host"] in user_data["subscriptions"]:
+                    recommended.append(item)
+    return recommended
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    most_genre = []
+    genre_recommended = []
+    users_movies = []
+    friends_movies = []
+    
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+        most_genre.append(movie['genre'])
+
+    if len(users_movies) == 0:
+        return friends_movies
+
+    max(set(most_genre), key=most_genre.count)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in genre_recommended:
+                if item["host"] in user_data["subscriptions"]:
+                    if most_genre[0] in item['genre']:
+                        genre_recommended.append(item)
+    return genre_recommended
+
+def get_rec_from_favorites(user_data):
+    fav_recommended = []
+    users_movies = []
+    friends_movies = []
+    
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    if len(users_movies) == 0:
+        return friends_movies

```

Comment:
Error checking here is great, but it's not necessary to build a new list to do it.  Also, returning `friends_movies` works because it's currently the empty list, but from a readability standpoint it's not clear that the goal is to return the empty list.
```suggestion
    if len(user_data["watched"]) == 0:
        return []

```